pkg/ext-proc/scheduling/filter.go

```diff
@@ -88,32 +88,48 @@ func toFilterFunc(pp podPredicate) filterFunc {
 	}
 }
 
+// leastQueuingFilterFunc finds the max and min queue size of all pods, divides the whole range
+// (max-min) by the number of pods, and finds the pods that fall into the first range.
 func leastQueuingFilterFunc(b *LLMRequest, pods []*backend.PodMetrics) ([]*backend.PodMetrics, error) {
 	min := math.MaxInt
+	max := 0
 	filtered := []*backend.PodMetrics{}
+
 	for _, pod := range pods {
-		if pod.WaitingQueueSize < min {
+		if pod.WaitingQueueSize <= min {
 			min = pod.WaitingQueueSize
-			filtered = []*backend.PodMetrics{}
 		}
-		if pod.WaitingQueueSize == min {
+		if pod.WaitingQueueSize >= max {
+			max = pod.WaitingQueueSize
+		}
+	}
+
+	for _, pod := range pods {
+		if pod.WaitingQueueSize >= min && pod.WaitingQueueSize <= min+(max-min)/len(pods) {
 			filtered = append(filtered, pod)
 		}
 	}
 	return filtered, nil
 }
 
+// leastKVCacheFilterFunc finds the max and min KV cache of all pods, divides the whole range
+// (max-min) by the number of pods, and finds the pods that fall into the first range.
```

Comment:
What is the intuition behind setting the range this way compared to finding the least-k elements for example?

Reply:
Updated the comment with the reason behind this. And added a TODO to compare with other strategies.